src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -406,6 +409,82 @@ private BoundStatement BindGoto(GotoStatementSyntax node, DiagnosticBag diagnost
             }
         }
 
+        private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax node, DiagnosticBag diagnostics)
+        {
+            // already defined symbol in containing block
+            var localSymbol = this.LookupLocalFunction(node.Identifier);
+
+            var hasErrors = false;
+
+            // In error scenarios with misplaced code, it is possible we can't bind the local declaration.
+            // This occurs through the semantic model.  In that case concoct a plausible result.
+            if (localSymbol == null)
+            {
+                localSymbol = new LocalFunctionSymbol(this, this.ContainingType, this.ContainingMemberOrLambda, node);
+            }
+            else
+            {
+                hasErrors |= this.ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
+            }
+
+            localSymbol.GrabDiagnostics(diagnostics);
+
+            var binder = this.GetBinder(node);
+
+            // Binder could be null in error scenarios (as above)
+            BoundBlock block;
+            if (binder != null)
+            {
+                if (node.Body != null)
+                {
+                    block = binder.BindBlock(node.Body, diagnostics);
+                }
+                else if (node.ExpressionBody != null)
+                {
+                    block = binder.BindExpressionBodyAsBlock(node.ExpressionBody, diagnostics);
+                }
+                else
+                {
+                    block = null;
+                    hasErrors = true;
+                    // TODO: add a message for this? (but parser currently doesn't produce syntax node with both null)
+                    diagnostics.Add(ErrorCode.ERR_ConcreteMissingBody, localSymbol.Locations[0], localSymbol);
+                }
+
+                if (block != null)
+                {
+                    // Have to do ControlFlowPass here because in MethodCompiler, we don't call this for synthed methods
+                    // rather we go directly to LowerBodyOrInitializer, which skips over flow analysis (which is in CompileMethod)
+                    // (the same thing - calling ControlFlowPass.Analyze in the lowering - is done for lambdas)
+                    var endIsReachable = ControlFlowPass.Analyze(localSymbol.DeclaringCompilation, localSymbol, block, diagnostics);
```

Comment:
Not exactly, see the comment. We can't re-use FlowAnalysisPass.Rewrite as there's some lazy computation there specific to normal method compiling and it performs too many diagnostics than what we want here, and the other use of ControlFlowPass.Analyze is done during UnboundLambda binding, which is difficult to extract/re-use because of use of its delegate type.

While yes, it's possible to rip out and refactor either of those two, I feel like the refactoring would complicate code more than re-use gains. I'll add a note in the code if I think of something clever later.